packages/protocol/src/action-protocol/clipboard.ts

```diff
@@ -14,80 +14,154 @@
-        return new RequestClipboardDataAction(editorContext);
+    export function is(object: any): object is RequestClipboardDataAction {
+        return RequestAction.hasKind(object, KIND) && hasObjectProp(object, 'editorContext');
     }
-}
 
-export function isRequestClipboardDataAction(action: any): action is RequestClipboardDataAction {
-    return isActionKind(action, RequestClipboardDataAction.KIND) && isObject(action, 'editorContext') && isString(action, 'requestId');
+    export function create(editorContext: EditorContext, options: { requestId?: string } = {}): RequestClipboardDataAction {
+        return {
+            kind: KIND,
+            requestId: '',
+            editorContext,
+            ...options
+        };
+    }
 }
 
 /**
  * Server response to a {@link RequestClipboardDataAction} containing the selected elements as clipboard-compatible format.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `SetClipboardDataActions`.
  */
-export class SetClipboardDataAction implements ResponseAction {
-    static readonly KIND = 'setClipboardData';
-
-    constructor(
-        public readonly clipboardData: ClipboardData,
-        public readonly responseId: string = '',
-        public readonly kind: string = SetClipboardDataAction.KIND
-    ) {}
+export interface SetClipboardDataAction extends ResponseAction {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof SetClipboardDataAction.KIND;
+
+    /**
+     * The selected elements from the editor context as clipboard data.
+     */
+    clipboardData: ClipboardData;
 }
 
-export function isSetClipboardDataAction(action: any): action is SetClipboardDataAction {
-    return isActionKind(action, SetClipboardDataAction.KIND) && isObject(action, 'clipboardData') && isString(action, 'responseId');
+export namespace SetClipboardDataAction {
+    export const KIND = 'setClipboardData';
+
+    export function is(object: any): object is SetClipboardDataAction {
+        return Action.hasKind(object, KIND) && hasObjectProp(object, 'clipboardData');
+    }
+
+    export function create(clipboardData: ClipboardData, options: { responseId?: string } = {}): SetClipboardDataAction {
+        return {
+            kind: KIND,
+            responseId: '',
+            clipboardData,
+            ...options
+        };
+    }
 }
 
 /**
  * Requests a cut operation from the server, i.e., deleting the selected elements from the model. Before submitting a `CutOperation`
  * a client should ensure that the cut elements are put into the clipboard.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `CutOperations`.
  */
-export class CutOperation implements Operation {
-    static readonly KIND = 'cut';
+export interface CutOperation extends Operation {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.